include/photos.php

```diff
@@ -42,16 +45,28 @@ function photo_albums($uid, $update = false) {
 	$key = "photo_albums:".$uid.":".local_user().":".remote_user();
 	$albums = Cache::get($key);
 	if (is_null($albums) OR $update) {
-		/// @todo This query needs to be renewed. It is really slow
-		// At this time we just store the data in the cache
-		$albums = qu("SELECT COUNT(DISTINCT `resource-id`) AS `total`, `album`
-			FROM `photo` USE INDEX (`uid_album_created`)
-			WHERE `uid` = %d  AND `album` != '%s' AND `album` != '%s' $sql_extra
-			GROUP BY `album` ORDER BY `created` DESC",
-			intval($uid),
-			dbesc('Contact Photos'),
-			dbesc(t('Contact Photos'))
-		);
+		if (!Config::get('system', 'no_count', false)) {
+			/// @todo This query needs to be renewed. It is really slow
```

Comment:
Just to share my thoughts:
When a picture is uploaded the cache of the profile owner should be updated because he/she is the one who owns all privileges. If a visitor comes to the photo page we could check if there is already a cache entry for the visitor. If not we create a new cache entry for the visitor in respect to his/her permissions (this is the way like it is working already). If there exists already a cache entry for the visitor we could check if the cache entry of the owner is newer (because this entry should have the most recent data because his/her entry should be updated on upload) than the entry of the visitor. If this is the case this would mean that the cached data of the visitor seems to be outdated and would need a fresh update. This would appear in respect to the permissions the visitor does own (because we already doing it this way).

There may be the case that the owner will upload pictures marked with private access permissions. Sure in this case the cache entry of visitor will also updated although there is no new content according to his/her permission level. But this would also happen if this user haven't visit the photo page for 24h.

It's more a compromise between of having possible recent data (and don't have to wait 24h until the cache gets deleted) and performance (does mean using the cache)